pkg/cleaner/azure/cleaner.go

```diff
@@ -36,5 +61,41 @@ func (c *Cleaner) Clean(ctx context.Context) error {
 	//     https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworkPeeringsClient
 	//
 
+	r, err := c.virtualNetworksClient.ListAll(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, v := range r.Values() {
+		// We skip all virtual networks of installations we do not want to cleanup.
+		if !containsString(c.installations, *v.Name) {
```

Comment:
I think it's better to select resources to delete than resources to reman. Just in case we don't configure new installation name on time.

All CI leftovers are created in godsmack resource group. And they are prefixed with `ci-`. Check peerings (which soon will be replaced by vnet gateways btw): https://portal.azure.com/#@accountsgiantswarm.onmicrosoft.com/resource/subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/godsmack/providers/Microsoft.Network/virtualNetworks/godsmack/peerings

Reply:
> I think it's better to select resources to delete than resources to remain. Just in case we don't configure new installation name on time.

I am not sure what you mean with this. What should I change here? What is the problem?

Reply:
I mean checking only godsmack (which is our only test installation) vnet, rather than skipping other non-test installations.

I also noticed you don't page over results.

Reply:
I am still trying to figure out how to list all the vnet peerings. The clients are super complex and rather confusing. I can make the selection explicit as soon as I know how. This here is all just WIP. I will merge this now and iterate from here. Next PR is coming. Also, when I do `ListAll` I should get all items and should not need to page.

Reply:
https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworkListResultPage.NotDone